app/src/main/java/io/appium/uiautomator2/utils/ScreenshotHelper.java

```diff
@@ -147,17 +158,51 @@ public static String takeScreenshot() throws TakeScreenshotException {
         }
     }
 
+    public static byte[] compressJpeg(final Bitmap bitmap, float scale, int quality, boolean filter) throws TakeScreenshotException {
```

Comment:
As far as I can read in Android documentation Jpeg compression is the most resource-intensive operation in general. What if we try to use the low level MediaCodec interfaces to produce Mjpeg stream: https://github.com/archersmind/EncoderDecoderTestForMJPEG/blob/master/EncoderDecoderTestForMJPEG.java

Reply:
I think this seems like a great future optimization.

Reply:
Marking as resolved.